evidence/pool.go

```diff
@@ -511,19 +514,18 @@ func evMapKey(ev types.Evidence) string {
 	return string(ev.Hash())
 }
 
-// big endian padded hex
-func bE(h int64) string {
-	return fmt.Sprintf("%0.16X", h)
-}
-
 func keyCommitted(evidence types.Evidence) []byte {
-	return append([]byte{baseKeyCommitted}, keySuffix(evidence)...)
+	key, err := orderedcode.Append([]byte{prefixCommitted}, evidence.Height(), string(evidence.Hash()))
+	if err != nil {
+		panic(err)
+	}
+	return key
 }
 
 func keyPending(evidence types.Evidence) []byte {
-	return append([]byte{baseKeyPending}, keySuffix(evidence)...)
-}
-
-func keySuffix(evidence types.Evidence) []byte {
-	return []byte(fmt.Sprintf("%s/%X", bE(evidence.Height()), evidence.Hash()))
+	key, err := orderedcode.Append([]byte{prefixPending}, evidence.Height(), string(evidence.Hash()))
```

Comment:
We need to be careful with `int64` and `uint64` here, since changing the return value of `evidence.Height()` may cause the on-disk format to break. Might want to cast to `uint64` to make sure it's stable, unless `orderedcode` can decode them interchangably.

Reply:
What would you think about doing the opposite. Leaving it as is and when we change to uint64 we make sure we cast the height to int64. Int64 encoding has a different strategy to uint64 encoding which makes it almost always more or at least as efficient as uint64. 

If you want I could deliberately cast it to int64 now so whoever makes the change to uint64 should recognise that this needs to remain as int64

Reply:
`int64` can't store the full range of `uint64`. While overflow is unlikely, these sorts of impedance mismatches are a fertile source of bugs. I'd say either stick with `int64` while we're using `int64` for heights and then change it later, or go with `uint64` if that's the long-term plan.

Reply:
If/when we change from `int64` to `uint64` this will be a breaking change to the db regardless so would it not make sense to change it all then

Reply:
And yes, if we stick with `int64` we should make sure that we pass `int64` to the key (either by a type assertion or checked cast), so that it won't suddenly break if someone changes the `evidence.Height()` return type.